test/unittest/transport/test_dynamic_buffer.cpp

```diff
@@ -0,0 +1,124 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#include "libcyphal/transport/dynamic_buffer.hpp"
+
+#include <gmock/gmock.h>
+
+using cetl::type_id_type;
+using cetl::rtti_helper;
+
+using DynamicBuffer = libcyphal::transport::DynamicBuffer;
+
+using testing::Return;
+using testing::StrictMock;
+
+namespace
+{
+
+class InterfaceMock : public DynamicBuffer::Interface
+{
+public:
+    MOCK_METHOD(void, moved, ());
+    MOCK_METHOD(void, deinit, ());
+
+    MOCK_METHOD(std::size_t, size, (), (const override));
+    MOCK_METHOD(std::size_t, copy, (const std::size_t, void* const, const std::size_t), (const override));
+};
+class InterfaceWrapper : public rtti_helper<type_id_type<0x01>, DynamicBuffer::Interface>
+{
+public:
+    explicit InterfaceWrapper(InterfaceMock* mock)
+        : mock_{mock}
+    {
+    }
+    InterfaceWrapper(InterfaceWrapper&& other) noexcept
+    {
+        move_from(std::move(other));
+    }
+    InterfaceWrapper& operator=(InterfaceWrapper&& other) noexcept
+    {
+        move_from(std::move(other));
+        return *this;
+    }
+    ~InterfaceWrapper() override
+    {
+        if (mock_ != nullptr)
+        {
+            mock_->deinit();
+            mock_ = nullptr;
+        }
+    }
+
+    // DynamicBuffer::Interface
+
+    CETL_NODISCARD std::size_t size() const override
+    {
+        return mock_ ? mock_->size() : 0;
+    }
+    CETL_NODISCARD std::size_t copy(const std::size_t offset_bytes,
+                                    void* const       destination,
+                                    const std::size_t length_bytes) const override
+    {
+        return mock_ ? mock_->copy(offset_bytes, destination, length_bytes) : 0;
+    }
+
+private:
+    InterfaceMock* mock_ = nullptr;
+
+    void move_from(InterfaceWrapper&& other)
+    {
+        mock_       = other.mock_;
+        other.mock_ = nullptr;
+
+        if (mock_ != nullptr)
+        {
+            mock_->moved();
+        }
+    }
+
+};  // InterfaceWrapper
+
+TEST(test_dynamic_buffer, move_ctor_assign__size)
```

Comment:
This might be a nitpick considering that this is happening in a test suite but identifiers containing double underscore are reserved in C and C++.